pkg/provider/azure_vmssflex_cache.go

```diff
@@ -181,18 +181,35 @@ func (fs *FlexScaleSet) getNodeVmssFlexID(nodeName string) (string, error) {
 		}
 		vmssFlexes := cached.(*sync.Map)
 
+		var vmssFlexIDs []string
 		vmssFlexes.Range(func(key, value interface{}) bool {
 			vmssFlexID := key.(string)
-			_, err := fs.vmssFlexVMCache.Get(vmssFlexID, azcache.CacheReadTypeForceRefresh)
-			if err != nil {
-				klog.Errorf("failed to refresh vmss flex VM cache for vmssFlexID %s", vmssFlexID)
+			vmssFlex := value.(*compute.VirtualMachineScaleSet)
+			vmssPrefix := *vmssFlex.Name
```

Comment:
use `to.String(vmssFlex.Name)` that could avoid panic

Reply:
Done